horovod/run/util/network.py

```diff
@@ -0,0 +1,43 @@
+import socket
+
+import psutil
+
+from horovod.run.util import threads
+
+
+def _get_local_host_addresses():
+    local_addresses = []
+    for intf_info_list in psutil.net_if_addrs().values():
+        for intf_info in intf_info_list:
+            if intf_info.family == socket.AF_INET:
+                local_addresses.append(intf_info.address)
+    return local_addresses
+
+
+def get_local_host_intfs():
+    return set(psutil.net_if_addrs().keys())
+
+
+def filter_local_addresses(host_names):
+    local_addresses = _get_local_host_addresses()
+
+    def resolve_host_name(host_name):
+        try:
+            return socket.gethostbyname(host_name)
+        except socket.gaierror:
+            return None
+
+    args_list = [[host] for host in host_names]
+    host_addresses = threads.execute_function_multithreaded(
+        resolve_host_name, args_list)
+
+    # host_addresses is a map
+    remote_host_names = []
+    for i in range(len(host_names)):
```

Comment:
More elegant:
```python
for host_name, host_address in zip(host_names, host_addresses):
```

Reply:
host_addresses is a map {int: addresses} that is why had to use the for loop.

Reply:
Ah, OK!